### DISCUSSION TITLE
How can I get a Model which has X relationship

### DISCUSSION THREAD

**Discussion by heyaj19** [spatie/laravel-query-builder#518] [Q&A]
I am trying to get all Bookings for a user - which is a polymorphic of bookable. So bookable can be user or guest

```
                      $bookings=Auth::user()->bookings()->getQuery()->with('managerSpeciality','bookable');

    $builtQuery=QueryBuilder::for($bookings)
       ->allowedFilters([
        AllowedFilter::scope('date_betweens'),
    ])
      ->get();

 ```

Currently I have two routes 
For detail
      Route::get('/bookings/{booking:slug}','Managersapi\Booking\BookingListingsController@show');
     
For all 
   Route::get('/bookings','Managersapi\Booking\BookingListingsController@list');

But I want to do something like 
Route::get('/bookings?bookable=User:12345','Managersapi\Booking\BookingListingsController@show');

How would something like this be possible?

**Comment by AlexVanderbist** [comment]
Hi! I'd suggest using a custom filter that uses the `whereHasMorph` method to filter the query based on the morph relationship. You can read more about the `whereHasMorph` method here: https://laravel.com/docs/master/eloquent-relationships#querying-polymorphic-relationships and more information about custom query builder filters here: https://spatie.be/docs/laravel-query-builder/v3/features/filtering#custom-filters

Good luck!
